### PR TITLE
Add immediate resolution proposals and support absolute approval/reject values

### LINKED ISSUE #1844: Immediate, single vote approval for proposals
will be used for join circle requests

### PR DESCRIPTION
**Pull request by danielr18**
Closes #1844

### CODE REVIEW

File: functions/src/proposals/business/proposals/calculateFinalState.ts [hunk 1]
```diff
@@ -9,22 +9,38 @@ interface CalculateFinalStateArgs {
...
+    const { weightedApproved, weightedRejected, total, totalMembersWithVotingRight, rejected, approved } = votes;
+    const { maxReject, minApprove, quorum, minApproveUnit, maxRejectUnit } = global
 
     if (!totalMembersWithVotingRight) return PROPOSAL_STATE.FAILED
 
     if (minApprove === 0 && quorum === 0) {
-        if (maxReject > 0 && weightedRejected <= maxReject) return PROPOSAL_STATE.PASSED
-        if (maxReject > 0 && weightedRejected > maxReject) return PROPOSAL_STATE.FAILED
         if (maxReject === 0) return PROPOSAL_STATE.PASSED
+        if (maxRejectUnit === 'percent') {
```
Comment (ariel-dao) [inline functions/src/proposals/business/proposals/calculateFinalState.ts]: percent as enum in all PR

File: functions/src/proposals/business/proposals/calculateFinalState.ts [hunk 2]
```diff
@@ -9,22 +9,38 @@ interface CalculateFinalStateArgs {
...
+            if (maxReject > 0 && weightedRejected <= maxReject) return PROPOSAL_STATE.PASSED
+            if (maxReject > 0 && weightedRejected > maxReject) return PROPOSAL_STATE.FAILED
+        } else if (maxRejectUnit === 'absolute') {
```
Comment (ariel-dao) [inline functions/src/proposals/business/proposals/calculateFinalState.ts]: enum

**Review by ariel-dao — CHANGES_REQUESTED** [review]
Looks ok, formatting makes it really tough to see changes, we'll format the entire project after this PR is merged.

This PR is quite dangerous even though it seems ok, please import production db locally and run the migrations to make sure the data looks valid.

Also, did you manage to test the flows with the new settings locally? It's pretty important and I don't think this one should be done on dev as it would be easy to miss some critical potential errors there

File: functions/src/proposals/services/proposals/createAssignCircle.ts [hunk 3]
```diff
@@ -67,7 +67,8 @@ export const createAssignCircle = async (payload: AssignCircleArgs) => {
             type: PROPOSALS.ASSIGN_CIRCLE,
             global,
             local,
-            limitations
+            limitations,
+            resolutionType: ProposalResolutionType.WAIT_FOR_EXPIRATION,
```
Comment (ariel-dao) [inline functions/src/proposals/services/proposals/createAssignCircle.ts:71]: Need to update related endpoints on Postman

File: functions/src/proposals/business/proposals/calculateFinalState.ts (hunk 1 above)
Comment (danielr18) [reply functions/src/proposals/business/proposals/calculateFinalState.ts]: Done

File: functions/src/proposals/business/proposals/calculateFinalState.ts (hunk 2 above)
Comment (danielr18) [reply functions/src/proposals/business/proposals/calculateFinalState.ts]: Done

File: functions/src/proposals/services/proposals/createAssignCircle.ts (hunk 3 above)
Comment (danielr18) [reply functions/src/proposals/services/proposals/createAssignCircle.ts:71]: How should I update those?

**Comment by danielr18** [comment]
I've documented the process of importing the production DB.
The migration worked as expected.

File: functions/src/proposals/services/proposals/createAssignCircle.ts (hunk 3 above)
Comment (ariel-dao) [reply functions/src/proposals/services/proposals/createAssignCircle.ts:71]: Go into our postman collection (you are already a part of it and change related endpoint payloads if needed - proposals probably only)

**Comment by ariel-dao** [comment]
deployed to dev

**Pull request merged**